lib/grape/dsl/inside_route.rb

```diff
@@ -29,7 +29,10 @@ module PostBeforeFilter
         def declared(params, options = {}, declared_params = nil)
           options = options.reverse_merge(include_missing: true, include_parent_namespaces: true)
 
-          declared_params ||= (!options[:include_parent_namespaces] ? route_setting(:declared_params) : (route_setting(:saved_declared_params) || [])).flatten(1) || []
+          all_declared_params = route_setting(:declared_params)
+          current_namespace_declared_params = route_setting(:saved_declared_params).last
```

Comment:
Something can be broken, because you have removed that `route_setting(:saved_declared_params) || []` piece.